### PR TITLE
[CI] Configure build/publishing of VSCode extension

### PR DESCRIPTION
**Pull request by darionhaase**
This builds the VSCode extension and adds the vsix-package as an asset to the GitHub release.
When a tagged release is published, the extension is published to the VSCode Marketplace (similar to publishing of Docker images to the registry).

As I find the handling of pre-releases in the VSCode marketplace strange (https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions), e.g. not compatible with semver, and I don't know how we should handle this, there is currently no support for automatic publishing of 'nightly' versions of the extension. The workaround for this is to manually download and install the extension from the GitHub release. But TBH I believe there is currently no need for nightly releases of the extension, so this should be fine.

Also restructured workflow files. Unfortunately, GitHub Workflows cannot be placed into subfolders. As a workaround I used # as a separator.

### CODE REVIEW

File: .github/workflows/build#vsix#publish_to_marketplaces.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        working-directory: ${{ env.EXTENSION_DIRECTORY }}
+        run: npm run package
+
+      - name: Align package filename with release version
+        working-directory: ${{ env.EXTENSION_DIRECTORY }}
+        run: |
+          PACKAGE_VERSION="$(
+            cat package.json | jq '.version' --raw-output
+          )"
+          mv caesar-${PACKAGE_VERSION}.vsix caesar-${{ inputs.version }}.vsix
```
Comment (Philipp15b) [inline .github/workflows/build#vsix#publish_to_marketplaces.yml:44]: Can you explain why you move the package here? Shouldn’t the version always agree with the tag?

File: .github/workflows/setup_gh_release.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+      - name: Get the release version from the tag
+        if: env.VERSION == ''
+        run: echo "VERSION=${{ inputs.git_ref }}" >> $GITHUB_ENV
+
+      - name: Show the version
+        run: |
+          echo "version is: $VERSION"
+
+      - name: Check that tag version and Cargo.toml version are the same
```
Comment (Philipp15b) [inline .github/workflows/setup_gh_release.yml:39]: Do these checks also run for tagged releases? I don’t see how right now.

**Comment by darionhaase** [comment]
Now also includes publishing to OpenVSX. Files have been renamed to use `-` instead of `#` Should be ready to merge now :)

**Review by Philipp15b — APPROVED** [review]

**Comment by Philipp15b** [comment]
Thanks!

**Pull request merged**